mm2src/coins/qrc20/qrc20_tests.rs

```diff
@@ -689,6 +689,65 @@ fn test_sender_trade_preimage_with_allowance() {
     assert_eq!(actual, expected);
 }
 
+#[test]
+fn test_get_sender_trade_fee_preimage_for_correct_ticker() {
+    // where balance for tQTUM is at 0 for the priv_key below (using 0 bal just to make sure we get required(TradePreimageError::NotSufficientBalance) result for get_sender_trade_fee)
+    let priv_key = [
+        48, 88, 65, 23, 20, 154, 63, 74, 243, 8, 108, 134, 154, 199, 60, 197, 51, 238, 7, 68, 199, 14, 127, 221, 89,
+        80, 37, 174, 221, 178, 233, 65,
+    ];
+    let conf = json!({
+        "coin":"QRC20",
+        "required_confirmations":0,
+        "pubtype":120,
+        "p2shtype":110,
+        "wiftype":128,
+        "segwit":true,
+        "mm2":1,
+        "mature_confirmations":2000,
+        "protocol": {
+        "type": "QRC20",
+        "protocol_data": {
+        "platform": "tQTUM",
+        "contract_address": "0xd362e096e873eb7907e205fadc6175c6fec7bc44"
+      }
+    }
+    });
+    let req = json!({
+        "method": "electrum",
+        "servers": [{"url":"electrum1.cipig.net:10071"}, {"url":"electrum2.cipig.net:10071"}, {"url":"electrum3.cipig.net:10071"}],
+        "swap_contract_address": "0xba8b71f3544b93e2f681f996da519a98ace0107a",
+    });
+
+    let contract_address = "0xd362e096e873eb7907e205fadc6175c6fec7bc44".into();
+    let ctx = MmCtxBuilder::new().into_mm_arc();
+    let params = Qrc20ActivationParams::from_legacy_req(&req).unwrap();
+
+    let coin = block_on(qrc20_coin_from_conf_and_params(
+        &ctx,
+        "QRC20",
+        "tQTUM",
+        &conf,
+        &params,
+        &priv_key,
+        contract_address,
+    ))
+    .unwrap();
+
+    let actual = block_on(coin.get_sender_trade_fee(TradePreimageValue::Exact(0.into()), FeeApproxStage::OrderIssue))
+        .err()
+        .unwrap()
+        .get_inner()
+        .to_string();
+    // expecting TradePreimageError::NotSufficientBalance
+    let expected = TradePreimageError::NotSufficientBalance {
+        coin: "tQTUM".to_string(),
+        available: BigDecimal::from_str("0").unwrap(),
+        required: BigDecimal::from_str("0.08").unwrap(),
+    };
+    assert_eq!(expected.to_string(), actual.to_string());
```

Comment:
I'd suggest comparing two `TradePreimageError` errors instead of strings.
In order to do it, you need to add `PartialEq` to the derivings.

Reply:
https://github.com/KomodoPlatform/atomicDEX-API/pull/1289#issuecomment-1143561836

Yes, it would be great! Thank you 🙂